packages/components/src/form-file-upload/index.tsx

```diff
@@ -50,9 +50,15 @@ export function FormFileUpload( {
 	// @todo: Temporary fix a bug that prevents Chromium browsers from selecting ".heic" files
 	// from the file upload. See https://core.trac.wordpress.org/ticket/62268#comment:4.
 	// This can be removed once the Chromium fix is in the stable channel.
-	const compatAccept = !! accept?.includes( 'image/*' )
-		? `${ accept }, image/heic, image/heif`
-		: accept;
+	// Prevent Safari from adding "image/heic" and "image/heif" to the accept attribute.
+	const isSafari =
+		globalThis.navigator?.userAgent.includes( 'Safari' ) &&
```

Comment:
You know what, if I look closely at the [Node docs](https://nodejs.org/api/globals.html#navigator), there is a small range of versions where `navigator` exists (>= v21) but `navigator.userAgent` doesn't (>= v21.1.0) 😂 So maybe optional chain that, too. This kind of edge case may already be a reason to prefer approaches that are safer in the general case (e.g. `useEffect`).

Reply:
Updated in https://github.com/WordPress/gutenberg/pull/67139/commits/460ece427aaa41c0649b657c6e661e58ca0fb267